### ISSUE TITLE
Manual: $ is used instead of # for commands that must be run as root

### ISSUE THREAD

**Issue by NotaseCretagen** [NixOS/nixpkgs#15073]
## Issue description
[This part of the manual](https://nixos.org/releases/nixos/unstable/nixos-16.09pre79453.32b7b00/manual/index.html#sec-changing-config
) says:

`WarningThese commands must be executed as root, so you should either run them from a root shell or by prefixing them with sudo -i.`

and yet the commands are prefixed with a `$` (without a `sudo -i`) instead of a `#`, eg.
`$ nixos-rebuild switch`
`$ nixos-rebuild boot`

I figure I must be missing something, hence why this isn't a pull request.

**Comment by joachifm** [comment]
The NixOS manual does not follow that convention, but uses `$` consistently throughout. I'm unsure whether using the `#` convention was ever considered.

**Comment by wlhlm** [comment]
I've recently into NixOS and coming from Archlinux, I was slightly confused by the manual as to which commands require root and which commands can be invoked by normal users. The ArchWiki differentiates pretty consistently between `#` (run command as root) and `$` (run command as normal user), which makes things easier.

Here is an excerpt from the [wiki editing guidelines](https://wiki.archlinux.org/index.php/Help:Style#Command_line_text) which I find sensible:

> - When using inline code […], no prompt symbol is to be represented, but any notes about permissions must be added explicitly to the surrounding text.
> - When using block code […], use `$` as a prompt for regular user commands; use `#` as a prompt for root commands.

>> Note: Because `#` is also used to denote comments in text files, you should always make sure to avoid ambiguities, usually by explicitly writing to run the command or edit a text file.

> - The sentence introducing a command block should usually end with `:`.
> - Unless really necessary, prefer using:
```
# command
```
instead of writing:
```
$ sudo command
```
> - Do not use the root prompt and the sudo command together, like in:
```
# sudo command
```
> The only exception is when sudo is used with the `-u` flag: in this case the prompt can match the others of the same code block, or default to `$`.

> - Do not add comments in the same box of a command, for example:
```
# pacman -S foo  #Install package foo
```
> - Do not assume the user uses sudo or other privilege escalation utilities (e.g. gksu, kdesu).

I don't know if there are explicit style guidelines for the NixOS manual, but if the above rules are acceptable, I can try to work on a PR based on them.

**Comment by joachifm** [comment]
@wlhlm adding a brief section near the beginning that establishes notation sounds like a good idea. For now, I think it suffices to introduce `$` and `#` as short-hands for user/root. It's diomatic and I think it's a clear improvement over the visually noisy warning boxes used currently. (Also: the more policy you want to add, the more we'd need to talk about it before implementing it, I think ... starting small is probably a smart move)

**Comment by wlhlm** [comment]
PR in #15589.

**Comment by wlhlm** [comment]
This can be closed now that #15589 got merged.
